### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by armandocomellas1**
### Hi team! the main purpose of this pull request is:

1.  Number one and most crucial proof l am capable of showing up that l learned react-redux and how to use it and implement some features.

2. Link to my Loom video: https://www.loom.com/share/f401edd474af4a4a9fe8846e563ae7fe
3. Link to my Live server project: https://meek-cucurucho-0a7b75.netlify.app/details

Learning Objectives:

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions and reducers in React.

Thank you for your effort to look to my code review.

💯

### CODE REVIEW

File: src/__test__/Main.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  it('Test Main component renders', () => {
+    const tree = renderer
+      .create(
+        <Provider store={store}>
+          <MainStat />
+        </Provider>,
+      );
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (smunozmo) [inline src/__test__/Main.test.js:17]: - [x] You have done an excellent job, this projector will look great in your portfolio! 😎 However, there are still missing some of the required tests for the activity, I would suggest adding all the required tests:

Screen Shot 2022-10-05 at 10 59 09 PM

File: src/components/Details.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      qualityAir = 'Fair';
+    } else if (sumPollution.sumNo2 <= 200 && sumPollution.sumPm10 <= 90 && sumPollution.sumO3 <= 180) {
+      qualityAir = 'Moderate';
+    } else if (sumPollution.sumNo2 <= 400 && sumPollution.sumPm10 <= 180 && sumPollution.sumO3 <= 240) {
+      qualityAir = 'Poor';
+    } else if (sumPollution.sumNo2 > 400 && sumPollution.sumPm10 > 180 && sumPollution.sumO3 > 240) {
+      qualityAir = 'Very Poor';
+    } else {
+      qualityAir = 'Very Poor';
+    }
```
Comment (smunozmo) [inline src/components/Details.js:72]: - Awesome feature! 🔥

**Review by smunozmo — CHANGES_REQUESTED** [review]
Hi @armandocomellas1, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 👏 

- Another cool project for your portfolio, congrats on it! 🎉 
- You did an awesome work with the styling! 🔥 
- Excellent job adding the live demo and video. 🤓 
- Home and Detail links are working correctly, great! ✅ 
- Well done using Redux to manage the state and API data, perfect! 😎 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! 🤓 

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

⚠️  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/Main.test.js (hunk 1 above)
Comment (armandocomellas1) [reply src/__test__/Main.test.js:17]: Thank you so much for this observation.

they are done! Mock connection to store and API.

File: src/components/Details.js (hunk 2 above)
Comment (armandocomellas1) [reply src/components/Details.js:72]: Yes, I did it because l wanted to know for real the quality of the air of each country of us.

**Review by Ismailco — APPROVED** [review]
### Hi @armandocomellas1, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

cr-approve

## Highlights

✔️  No linter errors
✔️ Used Gitfow
✔️ Work documented professionally

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by smunozmo** [comment]
@armandocomellas1 Great work! Ready for your portfolio! 🚀 🚀 🚀

**Pull request merged**